quill-core/src/test/scala/io/getquill/quotation/DynamicQuerySpec.scala

```diff
@@ -0,0 +1,589 @@
+      val d = {
+        val d = q.dynamic
+        (d: DynamicActionReturning[TestEntity, Int])
+      }
+      d.q mustEqual q
+    }
+  }
+
+  "query" - {
+
+    def test[T: QueryMeta](d: Quoted[Query[T]], s: Quoted[Query[T]]) =
+      testContext.run(d).string mustEqual testContext.run(s).string
+
+    "dynamicQuery" in {
+      test(
+        dynamicQuery[TestEntity],
+        query[TestEntity]
+      )
+    }
+
+    "dynamicQuerySchema" - {
+      "no aliases" in {
+        test(
+          dynamicQuerySchema[TestEntity]("test"),
+          querySchema[TestEntity]("test")
+        )
+      }
+      "one alias" in {
+        test(
+          dynamicQuerySchema[TestEntity]("test", alias(_.i, "ii")),
+          querySchema[TestEntity]("test", _.i -> "ii")
+        )
+      }
+      "multiple aliases" in {
+        test(
+          dynamicQuerySchema[TestEntity]("test", alias(_.i, "ii"), alias(_.s, "ss")),
+          querySchema[TestEntity]("test", _.i -> "ii", _.s -> "ss")
+        )
+      }
+      "dynamic alias list" in {
+        val aliases = List[DynamicAlias[TestEntity]](alias(_.i, "ii"), alias(_.s, "ss"))
+        test(
+          dynamicQuerySchema[TestEntity]("test", aliases: _*),
+          querySchema[TestEntity]("test", _.i -> "ii", _.s -> "ss")
+        )
+      }
+      "path property" in {
+        case class S(v: String) extends Embedded
+        case class E(s: S)
+        test(
+          dynamicQuerySchema[E]("e", alias(_.s.v, "sv")),
+          querySchema[E]("e", _.s.v -> "sv")
+        )
+      }
+    }
+
+    "map" - {
+      "simple" in {
+        test(
+          dynamicQuery[TestEntity].map(v0 => quote(v0.i)),
+          query[TestEntity].map(v0 => v0.i)
+        )
+      }
+      "dynamic" in {
+        var cond = true
+        test(
+          dynamicQuery[TestEntity].map(v0 => if (cond) quote(v0.i) else quote(1)),
+          query[TestEntity].map(v0 => v0.i)
+        )
+
+        cond = false
+        test(
+          dynamicQuery[TestEntity].map(v0 => if (cond) quote(v0.i) else quote(1)),
+          query[TestEntity].map(v0 => 1)
+        )
+      }
+    }
+
+    "flatMap" - {
+      "simple" in {
+        test(
+          dynamicQuery[TestEntity].flatMap(v0 => dynamicQuery[TestEntity]),
+          query[TestEntity].flatMap(v0 => query[TestEntity])
+        )
+      }
+      "mixed with static" in {
+        test(
+          dynamicQuery[TestEntity].flatMap(v0 => query[TestEntity]),
+          query[TestEntity].flatMap(v0 => query[TestEntity])
+        )
+
+        test(
+          query[TestEntity].flatMap(v0 => dynamicQuery[TestEntity]),
+          query[TestEntity].flatMap(v0 => query[TestEntity])
+        )
+      }
+      "with map" in {
+        test(
+          dynamicQuery[TestEntity].flatMap(v0 => dynamicQuery[TestEntity].map(v1 => quote((unquote(v0), unquote(v1))))),
+          query[TestEntity].flatMap(v0 => query[TestEntity].map(v1 => (v0, v1)))
+        )
+      }
+      "for comprehension" in {
```

Comment:
Whoa! Please document that for comprehensions with dynamic queries need to be unquoted in the yield. That totally confused me.